client/consensus/slots/src/lib.rs

```diff
@@ -518,6 +520,11 @@ pub async fn start_slot_worker<B, C, W, SO, CIDP, CAW, Proof>(
 			continue
 		}
 
+		if !permission_resolver.resolve(slot_info.slot).await {
```

Comment:
Should this check be a little bit upper? I mean just after the getting the `slot_info`.

Reply:
I think that we don't want to get permission and skip because other conditions are not met - this could lead to the situation where other nodes might miss a chance to build a block.

Reply:
Ok, maybe you are right